hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/server/JournaledEditsCache.java

```diff
@@ -123,8 +125,14 @@ class JournaledEditsCache {
   JournaledEditsCache(Configuration conf) {
     float fraction = conf.getFloat(DFSConfigKeys.DFS_JOURNALNODE_EDIT_CACHE_SIZE_FRACTION_KEY,
         DFSConfigKeys.DFS_JOURNALNODE_EDIT_CACHE_SIZE_FRACTION_DEFAULT);
+    if (fraction <= 0 || fraction >= 1.0f) {
+      terminate(1, new IllegalArgumentException(String.format(
```

Comment:
`terminate()` seems too strong to me; I would expect that we just use `Preconditions.checkArgument()` to throw an exception. This is used elsewhere to validate config values, e.g. in `FSDirectory` constructor and `HAUtil.getNameNodeIdOfOtherNodes()`.

Reply:
Thanks @xkrogen for the review. I updated it.